librad/src/net/discovery/mdns.rs

```diff
@@ -34,42 +39,99 @@ impl Mdns {
         local_peer: PeerId,
         listen_addrs: Vec<SocketAddr>,
         query_interval: Duration,
+        cache_ttl: Duration,
     ) -> Result<Self, Box<dyn std::error::Error>> {
         let mut srv = MdnsService::new(true)?;
         srv.register(SERVICE_NAME);
         let query = srv.discover(SERVICE_NAME, query_interval);
 
         let ptr = format!("{}.{}", local_peer, SERVICE_NAME);
         let cname = format!("{}.radicle.local", local_peer);
+        let cache = Arc::new(RwLock::new(Discoveries::new(cmp::max(
+            cache_ttl,
+            query_interval,
+        ))));
+        let (tx, rx) = mpsc::channel(64);
```

Comment:
It's not `42`.

Reply:
Channel semantics remain mysterious to me: previously, this was of depth `1`, with the assumption that this would essentially apply backpressure (ie. block the DNS query loop, over which we have no control). Turns out, this looses data :shrug:

Reply:
This is surprising indeed as I'd assume the same semantic. It might be to due to the use of `SinkEt::send` instead of `Sender::try_send`.